Fixes to README and deploy files

sukanyaparashar opened the pull request:
This PR adds the fixes to README and also the deploy files to make the `main.go` file more cleaner and understandable.

sukanyaparashar committed:
Fixes to README and deploy files

mnedelchev-vn commented:
@sukanyaparashar I've tested the new tutorial way and we can improve some stuff:
1. Remove `@openzeppelin/contracts` folder and create `package.json` file with follow content:
`{
  "dependencies": {
    "@openzeppelin/contracts": "^5.0.2"
  }
}
`
Then before users run the `go run main.go` we have to add in the readme that they have run `npm i`. This way we remove the responsibility of hardcoding a specific library version and we always make sure that users always download the up to date openzeppelin library.
2. Please change the pragma versions of both contracts in folder `contracts` to `pragma solidity >=0.8.20;`. Why? Basically to reduce the cases where user is using a bit older solc version and he has to upgrade or downgrade his solc version.
3. Let's remove the `build` and `contractsgo` folders and please do include the commands for the building process in the readme. This is related to 1. - basically to make sure the user is always building his *.go files with latest `@openzeppelin/contracts` library. Here are the commands:
    - **Storage.sol**
        - `solc --abi --include-path node_modules/ --base-path ./ ./contracts/Storage.sol -o build`
        - `solc --bin --include-path node_modules/ --base-path ./ ./contracts/Storage.sol -o build`
        - `abigen --abi ./build/Storage.abi --pkg contractsgo --type Storage --out ./contractsgo/Storage.go --bin ./build/Storage.bin`
    - **TestERC20.sol**
        - `solc --abi --include-path node_modules/ --base-path . ./contracts/TestERC20.sol -o build`
        - `solc --bin --include-path node_modules/ --base-path . ./contracts/TestERC20.sol -o build`
        - `abigen --abi ./build/TestERC20.abi --pkg contractsgo --type TestERC20 --out ./contractsgo/TestERC20.go --bin ./build/TestERC20.bin`

sukanyaparashar committed:
Remove build folder, add openzeppelin in node_modules

sukanyaparashar committed:
Remove build folder and contents from contractsgo folder

sukanyaparashar committed:
Add empty contractsgo folder

sukanyaparashar committed:
Add .gitkeep in contractsgo folder

sukanyaparashar committed:
Fix README

sukanyaparashar commented:
> @sukanyaparashar I've tested the new tutorial way and we can improve some stuff:
> 
> 1. Remove `@openzeppelin/contracts` folder and create `package.json` file with follow content:
>    `{   "dependencies": { "@openzeppelin/contracts": "^5.0.2"   } } `
>    Then before users run the `go run main.go` we have to add in the readme that they have run `npm i`. This way we remove the responsibility of hardcoding a specific library version and we always make sure that users always download the up to date openzeppelin library.
> 2. Please change the pragma versions of both contracts in folder `contracts` to `pragma solidity >=0.8.20;`. Why? Basically to reduce the cases where user is using a bit older solc version and he has to upgrade or downgrade his solc version.
> 3. Let's remove the `build` and `contractsgo` folders and please do include the commands for the building process in the readme. This is related to 1. - basically to make sure the user is always building his *.go files with latest `@openzeppelin/contracts` library. Here are the commands:
>    
>    * **Storage.sol**
>      
>      * `solc --abi --include-path node_modules/ --base-path ./ ./contracts/Storage.sol -o build`
>      * `solc --bin --include-path node_modules/ --base-path ./ ./contracts/Storage.sol -o build`
>      * `abigen --abi ./build/Storage.abi --pkg contractsgo --type Storage --out ./contractsgo/Storage.go --bin ./build/Storage.bin`
>    * **TestERC20.sol**
>      
>      * `solc --abi --include-path node_modules/ --base-path . ./contracts/TestERC20.sol -o build`
>      * `solc --bin --include-path node_modules/ --base-path . ./contracts/TestERC20.sol -o build`
>      * `abigen --abi ./build/TestERC20.abi --pkg contractsgo --type TestERC20 --out ./contractsgo/TestERC20.go --bin ./build/TestERC20.bin`

@mnedelchev-vn I have changed the following according to your suggestions -

1. `@openzeppelin/contracts` is now installed with `npm install`.
2. Contracts are now using `pragma solidity >= 0.8.20`. And the `TestERC20.sol` is importing `OpenZeppelin` contracts from the `node_modules` folder.
3. Removed the `build` folder because it gets generated automatically when generating the ABI and bin files. But I have created an empty `contractsgo` folder because it doesn't gets created automatically when go-binding of the smart contracts is generated. It will remove one step for the user for creating a folder.
4. Fixed the `README` with the required commands.

Please have a look at it and let me know.

mnedelchev-vn reviewed (approved):
lgtm
